oak_restricted_kernel/src/lib.rs

```diff
@@ -204,7 +205,7 @@ pub fn start_kernel(info: &BootParams) -> ! {
             .as_ptr()
         };
         BASE_L4_PAGE_TABLE
-            .set((pml4.clone(), encrypted))
+            .set(pml4.clone())
```

Comment:
Strictly speaking it would not need to be cloned now. With this PR we'll switch to a new pagetable when starting the app, so we can safely use the `BASE_L4_PAGE_TABLE` until then, without filling application space into it.

I kept the clone because the type system required it (expecting an owned type), but we can work around that by getting a mut pointer above.